tests/register.spec.coffee

```diff
@@ -11,22 +11,32 @@ describe 'Device Register:', ->
 
 	describe '.generateUUID()', ->
 
-		it 'should return a string', ->
-			uuid = register.generateUUID()
-			expect(uuid).to.be.a('string')
-
-		it 'should have a length of 62 (31 bytes)', ->
-			uuid = register.generateUUID()
-			expect(uuid).to.have.length(62)
-
-		it 'should generate different uuids each time', ->
-			uuid1 = register.generateUUID()
-			uuid2 = register.generateUUID()
-			uuid3 = register.generateUUID()
-
-			expect(uuid1).to.not.equal(uuid2)
-			expect(uuid2).to.not.equal(uuid3)
-			expect(uuid3).to.not.equal(uuid1)
+		it 'should return a string in its callback', (done) ->
+			register.generateUUID (err, uuid) ->
+				expect(uuid).to.be.a('string')
+				done()
+
+		it 'should return a string if called in Promise mode', (done) ->
+			register.generateUUID()
+			.then (uuid) ->
+				expect(uuid).to.be.a('string')
+				done()
+
+		it 'should have a length of 62 (31 bytes)', (done) ->
+			register.generateUUID (err, uuid) ->
+				expect(uuid).to.have.length(62)
+				done()
+
+		it 'should generate different uuids each time', (done) ->
+			Promise.all([
+				register.generateUUID()
+				register.generateUUID()
+				register.generateUUID()
+			]).then ([ uuid1, uuid2, uuid3 ]) ->
+				expect(uuid1).to.not.equal(uuid2)
+				expect(uuid2).to.not.equal(uuid3)
+				expect(uuid3).to.not.equal(uuid1)
+				done()
```

Comment:
Can you use `.nodeify()` here? Otherwise a possible error from the promise will not be passed to `done`.